File: C/binary search/mistral-saba-24b.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int binarySearch(int arr[], int left, int right, int target) {
    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == target)
            return mid;

        if (arr[mid] < target)
            left = mid + 1;
        else
            right = mid - 1;
    }

    return -1;
}

int main() {
    srand(time(0));
    int n = 1000;
    int arr[n];
    for (int i = 0; i < n; i++) {
        arr[i] = rand() % 10000;
    }

    // Sort the array
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }

    // Perform binary search
    int target = rand() % 10000;
    int result = binarySearch(arr, 0, n - 1, target);

    return 0;
}

File: Python/binary search/mistral-saba-24b.py
import random

def binary_search(arr, target):
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2

        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1

# Generate 1000 random integers sorted in ascending order
random_inputs = sorted(random.randint(0, 10000) for _ in range(1000))

# Perform binary search on the sorted list
target = random.choice(random_inputs)
result = binary_search(random_inputs, target)